crates/sparrow-compiler/src/types/inference.rs

```diff
@@ -296,18 +369,54 @@ fn solve_constraint(
                         .with_message(format!("Invalid types for call to '{call}'")),
                 )
                 .with_labels(distinct_arg_types)
-                .with_note(format!("Expected '{constraint}'"))
+                .with_note(format!("Expected '{type_class}'"))
         })
 }
 
 /// Instantiate the (possibly generic) `FenlType` using the computed
 /// solutions.
 fn instantiate_type(fenl_type: &FenlType, solutions: &HashMap<TypeVariable, FenlType>) -> FenlType {
     match fenl_type {
-        FenlType::Generic(type_var) => solutions
+        FenlType::TypeRef(type_var) => solutions
             .get(type_var)
             .cloned()
             .unwrap_or(FenlType::Concrete(DataType::Null)),
+        FenlType::Collection(c, type_vars) => match c {
```

Comment:
Then down here, actually instantiating the types for the collection - for each type variable in the collection, I get the concrete fenl type. But what's weird is that I am manually assigning the field name to be `key` and `value`, as per: 

```
    /// A Map is a logical nested type that is represented as
    ///
    /// `List<entries: Struct<key: K, value: V>>`
    ///
    /// The keys and values are each respectively contiguous.
    /// The key and value types are not constrained, but keys should be
    /// hashable and unique.
    /// Whether the keys are sorted can be set in the `bool` after the `Field`.
    ///
    /// In a field with Map type, the field has a child Struct field, which then
    /// has two children: key type and the second the value type. The names of the
    /// child fields may be respectively "entries", "key", and "value", but this is
    /// not enforced.
    Map(FieldRef, bool),
```

But my actual type (which was done above) is a `DataType::Struct(fields: {f1: .., f2: ..})`. (Where `f1` and `f2` are the map keys, from the input:
```
    m                          map name                 time
0  f1     {'f1': 'dog', 'f2': 'cat'}    A  1996-12-19T16:39:57

```

Reply:
Which gets me to the error I'm at currently:
```
│
╰─▶ No field named 'value' found in struct Concrete(Struct([Field { name: "f1", data_type: Utf8, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "f2", data_type: Utf8, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }]))
    ├╴at crates/sparrow-compiler/src/compile.rs:97:10
    ├╴backtrace (1)
    ╰╴span trace with 1 frames (1)
```